lib/paramedic.js

```diff
@@ -190,48 +192,160 @@ ParamedicRunner.prototype.writeMedicConnectionUrl = function(url) {
 };
 
-ParamedicRunner.prototype.runTests = function () {
+ParamedicRunner.prototype.buildApp = function () {
     var self = this;
-    if (this.config.shouldUseSauce()) {
-        var command = this.getCommandForBuilding();
+    var command = this.getCommandForBuilding();
+
+    logger.normal('cordova-paramedic: running command ' + command);
+
+    return execPromise(command)
+    .fail(function(output) {
+        // this trace is automatically available in verbose mode
+        // so we check for this flag to not trace twice
+        if (!self.config.verbose) {
+            logger.normal(output);
+        }
+        logger.normal('cordova-paramedic: unable to build project; command log is available above');
+        throw new Error('Unable to build project.');
+    });
+};
+
+ParamedicRunner.prototype.runLocalTests = function () {
+    var self = this;
+
+    return self.getCommandForStartingTests()
+    .then(function(command) {
+        self.setPermissions();
         logger.normal('cordova-paramedic: running command ' + command);
 
-        return execPromise(command)
-        .then(self.runSauceTests.bind(self), function(output) {
-            // this trace is automatically available in verbose mode
-            // so we check for this flag to not trace twice
-            if (!self.config.verbose) {
-                logger.normal(output);
-            }
-            logger.normal('cordova-paramedic: unable to build project; command log is available above');
-            throw new Error('Command "' + command + '" failed.');
+        return execPromise(command);
+    })
+    .then(function() {
+        // skip tests if it was just build
+        if (self.shouldWaitForTestResult()) {
+            return Q.promise(function(resolve, reject) {
+                // reject if timed out
+                self.waitForConnection().catch(reject);
+                // resolve if got results
+                self.waitForTests().then(resolve);
+            });
+        }
+    }, function(output) {
+        // this trace is automatically available in verbose mode
+        // so we check for this flag to not trace twice
+        if (!self.config.verbose) {
+            logger.normal(output);
+        }
+        logger.normal('cordova-paramedic: unable to run tests; command log is available above');
+        throw new Error('Unable to run tests.');
+    });
+};
+
+ParamedicRunner.prototype.getAppiumDeviceName = function () {
+    if (!this.targetObj) {
+        return '';
+    }
+    if (this.config.getPlatformId() === 'ios') {
+        return this.targetObj.target.replace('-', ' ');
+    }
+    return this.targetObj.target;
+};
+
+ParamedicRunner.prototype.runAppiumTests = function () {
+    var platform = this.config.getPlatformId();
+    if (platform !== 'android' && platform !== 'ios') {
+        logger.warn('Unsupported platform for Appium test run: ' + platform);
+        return Q(true);
```

Comment:
What does `true` value mean for caller? Does this indicate that tests are completed or succeeded? Could you please add comment clarifying this?